### ISSUE TITLE
Drop support for JRE 8

### ISSUE THREAD

**Issue by ohecker** [devonfw/solicitor#250]
This is a follow up of #249 .

As the product owner of Solicitor I want to build Solicitor on/for Java 11 and without support for JRE 8.
This will then enable to update to more recent versions of used components. See e.g. https://github.com/devonfw/solicitor/issues/246.

JRE 8 should be dropped and The system should be built using Java 11.

**Comment by ohecker** [comment]
Follow up activites after dropping Support of JRE8:
- Update Drools to Version >= 8.28.0.Beta
- Remove workaround for https://issues.redhat.com/browse/DROOLS-7210 which was introduced in DroolsRuleEngine with https://github.com/devonfw/solicitor/commit/25be0d7c97b979942567d1c8eca839edd9558989 (should no longer be necessary is Drools >= 8.31.0.Beta is used.)

**Comment by psistorm** [comment]
I hope this could be the right place. Solicitor is not able to run with any JDK higher than 17. To fix this we just needed to update three dependencies

```
    <dependency>
      <groupId>org.kie</groupId>
      <artifactId>kie-ci</artifactId>
      <version>8.44.2.Final</version>
    </dependency>
    <dependency>
      <groupId>org.drools</groupId>
      <artifactId>drools-decisiontables</artifactId>
      <version>8.44.2.Final</version>
    </dependency>
    <dependency>
      <groupId>org.drools</groupId>
      <artifactId>drools-xml-support</artifactId>
      <version>8.44.2.Final</version>
    </dependency>
```

So it's basically updating 8.27.0.Beta for 8.44.2.Final

Any chance this could get included in the next release?

**Comment by ohecker** [comment]
Hi @psistorm : Thank you for your request and included proposed change. Am I right that the issue you encountered is due to some ClassNotFoundException of java/lang/Compiler ? If yes: java.lang.Compiler was deprecated with JDK 8 and was removed after JDK 17.  We currently still support JDK 8. This is now in "deprecation stage 2" (see #249) meaning that Solicitor will by default terminate execution on JDK 8 - but you can still force it to work by setting a specific property. 

We are checking if we can drop support for JDK 8 for the next release and include your change.

In case you were facing another problem with JDK 21 just drop a note.

**Comment by psistorm** [comment]
Yes this was exactly the problem "java.lang.Compiler" not found and it was fixed by updating the libs. Thanks for integrating this in the official build! This helped us relieve a major headache and avoiding a fork.
